maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,223 @@
+		global $wgStatsDB;
+
+		$dbw = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
+		return $dbw->selectFieldValues( 'city_used_tags', 'ct_page_id', [
+			'ct_wikia_id' => $cityId,
+			'ct_kind' => 'imap'
+		] );
+	}
+
+	public function hasImapTag( $articleId, &$foundTags, &$foundTagsMapIds ) {
+		$article = Article::newFromID( $articleId );
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$results = null;
+			$articleContent = $article->getContent();
+			$imapSearchRegexp = "/<imap.*map\-id\=['\"](\d{1,})['\"].*(<\/imap|\/)>/";
+
+			$noOfFoundTags = preg_match_all($imapSearchRegexp, $articleContent, $results);
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function hasTagsToRemove( $foundTagsMapIds, $mapsUsingTheTileset, &$tagsToRemoveArrayKeyIds ) {
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
+			}
+		}
+
+		return !empty( $tagsToRemoveArrayKeyIds );
+	}
+
+	public function removeImapTagFromArticle( $articleId, $stringWithTagToRemove ) {
+		$user = User::newFromName( 'WikiaBot' );
+		$article = Article::newFromID( $articleId );
+
+		if ( $user->getId() === 0 ) {
+			$this->output( 'Failed using WikiaBot user' . PHP_EOL );
+		}
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$oldContent = $article->getContent();
+			$newContent = str_replace( $stringWithTagToRemove, "", $oldContent );
+
+			$this->output( sprintf( 'Trying to edit article #%d', $articleId ) . PHP_EOL );
+			if( !self::isTest() ) {
+				$result = $article->doEdit(
+					$newContent,
+					wfMessage(
+						'realmap-deprecated-info',
+						'[[community:User_blog:DaNASCAT/Technical_Update:_August_29,_2016]]'
+					)->text(),
+					EDIT_FORCE_BOT,
+					false,
+					$user
+				);
+			} else {
+				$result = true;
+			}
+
+
+			if( $result ) {
+				$this->output( sprintf( "Removed <imap/> tags from article #%d", $articleId ) . PHP_EOL );
+			} else {
+				$this->output( sprintf( "Failed to update article #%d", $articleId ) . PHP_EOL );
+			}
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function isValidTilesSetId( $tilesSetId ) {
+		if ( !self::isNaturalNumber( $tilesSetId ) ) {
+			return false;
+		}
+
+		$res = $this->maps->getTileSet( $tilesSetId );
+
+		if ( !isset( $res['success'] ) || $res['success'] !== true ) {
+			$this->error( 'API call failure when looking for a tiles set #' . $tilesSetId . '.' . PHP_EOL );
+		}
+
+		$this->output( "Tiles' set #" . $tilesSetId . " found." . PHP_EOL );
+		return $res['content']->id == $tilesSetId;
+	}
+
+	public function execute() {
+		$this->app = F::app();
+		$this->mapsClientConfig = $this->app->wg->IntMapConfig;
```

Comment:
Does `$this->app->wg->IntMapConfig` contain the database credentials (i.e. is it the content of [InteractiveMapsConfig.json](https://github.com/Wikia/config/blob/dev/InteractiveMapsConfig.json))?